### ISSUE TITLE
add image load safety

### ISSUE THREAD

**Issue by trailerzone** [flutter/flutter#80653]
it's kind of a weird idea because most people can just do a try else case on images, but why not set a flag to true for image load safety, basically what happens is if the asset does not render, instead of throwing an exception, why not paint an asset that says file not found or use https://publicdomainvectors.org/en/free-clipart/Vector-illustration-of-image-did-not-load-icon/22266.html

the reason why i am saying this is because, newbies don't know what that means when an exception is thrown, or they might not know how to fix it right away. and sometimes assets just don't load because the file is not perfectly formatted yet, this can be an issue for people with older laptop. think accessibility for those who are less fortunate to have a newer system. 

oftentimes when an asset is not perfect it will throw an exception, that takes people with older systems more time because they have to stop the app, fix the code and then recompile, which xcode updated their compiler and it takes almost 2 minutes to compile if the user has an older system and has to fire back up the code due to the image throw exception, i think it is found in the asset_bundle.dart file. 

i tried editing it but i broke the code and had to delete and redownload flutter in order to fix it. i'll try again one of these days and send in a push request. but something as simple as returning an flutter drawn asset by default on exception would make the code a lot more accessible to people who work or go to school full time and don't have the resources to get a new system and also who are new to the platform. think of it as the null safety for images.  just make a flag where someone can set to ignore the safety, if they want to code in their own exception. thanks

**Comment by iapicca** [comment]
@trailerzone 
does this fit your needs?
https://flutter.dev/docs/cookbook/images/fading-in-images

**Comment by trailerzone** [comment]
> @trailerzone
> does this fit your needs?
> https://flutter.dev/docs/cookbook/images/fading-in-images
> 
> ![fading-in-asset-demo](https://user-images.githubusercontent.com/45637986/115126778-67514b80-9fda-11eb-9acf-f170a9a1fc13.gif)

not exactly, this is more so for things like an svg loaded locally or a png/gif loaded locally, so if you want to load a local asset that throws an error, the system would display a placeholder image instead of throwing the error. so for for local files only, this widget looks like it was made for waiting on network images? unless it can load a placeholder if a local asset throws an error/can't load?

**Comment by trailerzone** [comment]
it's this area of image_provider.dart that throws the error when a local png cannot load. 

  @protected
  Future<ui.Codec> _loadAsync(AssetBundleImageKey key, DecoderCallback decode) async {
    ByteData? data;
    // Hot reload/restart could change whether an asset bundle or key in a
    // bundle are available, or if it is a network backed bundle.
    try {
      data = await key.bundle.load(key.name);
    } on FlutterError { // this area, why can't the system return a placeholder asset instead start says this message rather than breaking the code? maybe there is an easier way but, it looks problematic for a ton of people. 
      PaintingBinding.instance!.imageCache!.evict(key);
      rethrow;
    }
    // `key.bundle.load` has a non-nullable return type, but might be null when
    // running with weak checking, so we need to null check it anyway (and
    // ignore the warning that the null-handling logic is dead code).
    if (data == null) { // ignore: dead_code
      PaintingBinding.instance!.imageCache!.evict(key);
      throw StateError('Unable to read data');
    }
    return await decode(data.buffer.asUint8List());
  }
}

**Comment by JagandeepBrar** [comment]
`Image.asset` has an error builder callback that can be defined. More information here: https://api.flutter.dev/flutter/widgets/ImageErrorWidgetBuilder.html

**Comment by trailerzone** [comment]
there is 


> [Image.asset](https://api.flutter.dev/flutter/widgets/Image/Image.asset.html) has an error builder callback that can be defined. More information here: https://api.flutter.dev/flutter/widgets/ImageErrorWidgetBuilder.html

yeah except i do not think that it works on AssetImage for when you are using it with boxdecoration.

**Comment by iapicca** [comment]
> > [Image.asset](https://api.flutter.dev/flutter/widgets/Image/Image.asset.html) has an error builder callback that can be defined. More information here: https://api.flutter.dev/flutter/widgets/ImageErrorWidgetBuilder.html
> 
> yeah except i do not think that it works on AssetImage for when you are using it with boxdecoration.

@trailerzone 
can you provide a [minimal reproducible code sample](https://stackoverflow.com/help/minimal-reproducible-example) to demonstrate that

**Comment by TahaTesser** [comment]
Hi @trailerzone 
Can you please provide your `flutter doctor -v` and a [minimal complete reproducible code sample](https://stackoverflow.com/help/minimal-reproducible-example) and `flutter run --verbose` logs running the minimal code sample?
Thank you

**Comment by pedromassangocode** [comment]
Without additional information, we are unfortunately not sure how to resolve this issue. 
We are therefore reluctantly going to close this bug for now. 
Please don't hesitate to comment on the bug if you have any more information for us; we will reopen it right away!  
Thanks for your contribution.

Could everyone who still has this problem please file a new issue with the exact description of what happens, logs, and the output of `flutter doctor -v`.
All system setups can be slightly different, so it's always better to open new issues and reference related issues.
